lib/withOnyx.js

```diff
@@ -19,7 +20,7 @@ function getDisplayName(component) {
     return component.displayName || component.name || 'Component';
 }
 
-export default function (mapOnyxToState) {
+export default function (mapOnyxToState, options = {}) {
```

Comment:
Until there are other options, a simple param is fine.
```suggestion
export default function (mapOnyxToState, shouldDelayUpdates = false) {
```

Reply:
I strongly argue against it, boolean params are confusing because they are not labeled. An options object future proofs while keeping things explicit.

Reply:
While I don't disagree with the points of your logic, I disagree with the results of that logic. If that logic was followed, then no method anywhere would ever have single params and they would all have one big arguments object.

What about moving it to `mapOnyxToState`?

Reply:
Robert C. Martin says it himself:

https://www.informit.com/articles/article.aspx?p=1392524

But Szymon didn't want the key inside mapOnyxToState because that would change how the necessary keys for rendering are discovered. I will change it as a single boolean for now.

Reply:
Uncle Bob rules only apply if it's Java code

Reply:
Why would it? it's just code and boolean functions. They get messy over time, there is plenty of arguments against them and I have also lived through long-lived codebases where they make everything hard to understand.

https://medium.com/@amlcurran/clean-code-the-curse-of-a-boolean-parameter-c237a830b7a3

https://understandlegacycode.com/blog/what-is-wrong-with-boolean-parameters/

In any case, not going to push this on you, just sharing my experience.